install_requirements.py

```diff
@@ -63,175 +65,196 @@ def python_is_compatible():
+    shutil.rmtree("pip-out/", ignore_errors=True)
+    dirs = glob.glob("cmake-out*/") + glob.glob("cmake-android-out/")
+    for d in dirs:
+        print(f"Cleaning {d}...")
+        shutil.rmtree(d, ignore_errors=True)
+    print("Done cleaning build artifacts.")
 
-# Parse options.
 
-EXECUTORCH_BUILD_PYBIND = ""
-CMAKE_ARGS = os.getenv("CMAKE_ARGS", "")
-CMAKE_BUILD_ARGS = os.getenv("CMAKE_BUILD_ARGS", "")
-USE_PYTORCH_NIGHTLY = True
+def main(args):
+    if not python_is_compatible():
+        sys.exit(1)
 
-args = sys.argv[1:]
-for arg in args:
-    if arg == "--pybind":
-        pass
-    elif arg in ["coreml", "mps", "xnnpack"]:
-        if "--pybind" in args:
-            arg_upper = arg.upper()
-            EXECUTORCH_BUILD_PYBIND = "ON"
-            CMAKE_ARGS += f" -DEXECUTORCH_BUILD_{arg_upper}=ON"
-        else:
-            print(f"Error: {arg} must follow --pybind")
-            sys.exit(1)
-    elif arg == "off":
-        if "--pybind" in args:
-            if EXECUTORCH_BUILD_PYBIND == "ON":
-                print("Cannot turnoff pybind option as it is already set.")
-                sys.exit(1)
+    # Parse options.
+
+    EXECUTORCH_BUILD_PYBIND = ""
+    CMAKE_ARGS = os.getenv("CMAKE_ARGS", "")
+    CMAKE_BUILD_ARGS = os.getenv("CMAKE_BUILD_ARGS", "")
+    USE_PYTORCH_NIGHTLY = True
+
+    parser = argparse.ArgumentParser(prog="install_requirements")
+    parser.add_argument(
+        "--pybind",
+        action="append",
+        nargs="+",
+        help="one or more of coreml/mps/xnnpack, or off",
+    )
+    parser.add_argument(
+        "--clean",
+        action="store_true",
+        help="clean build artifacts and pip-out instead of installing",
+    )
+    parser.add_argument(
+        "--use-pt-pinned-commit",
+        action="store_true",
+        help="build from the pinned PyTorch commit instead of nightly",
+    )
+    args = parser.parse_args(args)
+    if args.pybind:
+        # Flatten list of lists.
+        args.pybind = list(itertools.chain(*args.pybind))
+        if "off" in args.pybind:
+            if len(args.pybind) != 1:
+                raise Exception(
+                    f"Cannot combine `off` with other pybinds: {args.pybind}"
+                )
             EXECUTORCH_BUILD_PYBIND = "OFF"
         else:
-            print(f"Error: {arg} must follow --pybind")
-            sys.exit(1)
-
-    elif arg == "--clean":
-        print("Cleaning build artifacts...")
-        print("Cleaning pip-out/...")
-        shutil.rmtree("pip-out/", ignore_errors=True)
-        dirs = glob.glob("cmake-out*/") + glob.glob("cmake-android-out/")
-        for d in dirs:
-            print(f"Cleaning {d}...")
-            shutil.rmtree(d, ignore_errors=True)
-        print("Done cleaning build artifacts.")
-        sys.exit(0)
-    elif arg == "--use-pt-pinned-commit":
+            for pybind_arg in args.pybind:
+                if pybind_arg not in ["coreml", "mps", "xnnpack"]:
+                    continue
```

Comment:
raise exception here?

Reply:
whoops! I misinterpreted the original code after I'd made some modifications to it. will fix.